### ISSUE TITLE
Having a combination of "create queries" and "can't create queries" on joined tables exposes field IDs instead of field names

### ISSUE THREAD

**Issue by Tony-metabase** [metabase/metabase#37752]
_edit: the original issue wasn't a bug, it was a unintuitive behavior of the previous "Data access" permission. After we split it into "View data" and "Create queries", this is clearer. But we keep the bug that we are showing field IDs instead of field names in the notebook editor._

### Describe the bug

When you have a question consisting of 2 tables joined together one having `no self service` and another `unrestricted` will cause notebook editor to expose the `no self service`

### To Reproduce

1. Go to Admin -> People -> Add New Group -> Name it group_1
2. Go to Admin -> People -> Invite Someone -> Assign it to group_1  
3. Go to Admin -> Permissions -> Sample Database -> Set the All-Users to `no self service` -> Then set the People table to `no self service` and Orders table to `unrestricted`

<img width="1512" alt="image" src="https://github.com/metabase/metabase/assets/110378427/ec7f54c7-2328-4cca-ae55-dfc6afca94f0">

4. Go to New -> Collection -> Test -> Give view permissions to group_1

<img width="1512" alt="image" src="https://github.com/metabase/metabase/assets/110378427/9b34d8c7-a7f0-48df-bda1-d182249cb7c9">

5. Go New Question -> Sample Database -> Join Orders with Products -> Group by 2 columns from Orders and 2 columns from Products -> Save it in Test

<img width="1471" alt="image" src="https://github.com/metabase/metabase/assets/110378427/6faf8a38-4b19-4886-8472-1b7b628a594e">

6. Now login as the User you created which is assigned to group_1 -> Then open the above question and click on the Question Editor

<img width="1174" alt="image" src="https://github.com/metabase/metabase/assets/110378427/ac08169f-2598-4803-ae72-0702bb5b79b5">

7. Notice that even though the GUI removes the table name form the join you can still find it in the Group by

### Expected behavior

Do not expose table names I guess

### Logs

None that are relevant

### Information about your Metabase installation

```JSON
1.48.3 and master
```

### Severity

Depends on how people structure their table names this can be problematic and leaks information

**Comment by Tony-metabase** [comment]
Related to https://github.com/metabase/metabase/issues/37751

**Comment by metamben** [comment]
@Tony-metabase, I agree that `People - User → 48` instead of `People - User → State` looks strange, but I'm not sure what the right behavior is. Running the query reveals the field names in full, so we have to define how to display names there too. (With some work, authors can hide these names if they really wish by introducing custom columns for them.)

<img width="628" alt="image" src="https://github.com/metabase/metabase/assets/103100869/2c7dd4e4-152f-4a62-90c3-f5270734d016">

**Comment by metamben** [comment]
When the BE produces the metadata for the FE, it checks if the user can read the table, which is implemented by checking that the user has `unrestricted` `view-data` AND `query-builder` `create-query` access. @luizarakaki should we change this?

**Comment by perivamsi** [comment]
Downgrading to P2 as per https://metaboat.slack.com/archives/C04DN5VRQM6/p1721036711685639

**Comment by ranquild** [comment]
The fix should be the same as https://github.com/metabase/metabase/issues/37751

**Comment by metamben** [comment]
@Tony-metabase, is this still relevant? If yes, with what permission settings?

**Comment by metamben** [comment]
@Tony-metabase, @mngr, is this issue still relevant?

With the current permission settings, if the user os not allowed to see the `Products` table, they see this:

<img width="693" alt="Image" src="https://github.com/user-attachments/assets/379c6ba3-a304-4c88-afde-7f6ee17e778e" />

and running the query results in a permission error:

<img width="607" alt="Image" src="https://github.com/user-attachments/assets/404e2c1f-d69d-4916-816a-4324e2ee61ae" />

Do we want to change something here? If yes, how?

**Comment by Tony-metabase** [comment]
Yep in theory this is not relevant anymore since we don't expose information that the user doesn't have permissions on ... I opened a separate issue for the `/notebook` issue which reveals the notebook editor even if a user doesn't have permission to edit.

But I don't see anything getting leaked it's just notebook editor of a question which one shouldn't be able to reach. I guess we should show you don't have permission to do this. This can be closed as fixed and i opened an issue for the other one:

https://github.com/metabase/metabase/issues/58628
